pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -45,10 +46,33 @@ const (
 	preFilterStateKey = "PreFilter" + FitName
 )
 
+// nodeResourceStrategyTypeMap store different ScoringStrategy's args validate function and scorer function
+var nodeResourceStrategyTypeMap = map[config.StrategyType]scorer{
+	config.LeastAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return newLeastAllocationScorer(leastResourceScorer(resToWeightMap), resToWeightMap)
+	},
+	config.MostAllocated: func(args *config.NodeResourcesFitArgs) *resourceAllocationScorer {
+		resToWeightMap := resourcesToWeightMap(args.ScoringStrategy.Resources)
+		return newMostAllocationScorer(mostResourceScorer(resToWeightMap), resToWeightMap)
```

Comment:
remove the new function and do this:

return &resourceAllocationScorer{
		Name:               MostAllocatedName,
		scorer:              mostResourceScorer(resToWeightMap),
		resourceToWeightMap: resToWeightMap,
	}

Reply:
and for the other two